contracts/orbital-auction/src/msg.rs

```diff
@@ -0,0 +1,70 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{Coin, Uint128};
+
+use crate::state::{AuctionPhase, AuctionPhaseConfig, RouteConfig, UserIntent};
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    // config that describes the route:
+    // - src & destination domains
+    // - offer & ask denoms
+    pub route_config: RouteConfig,
+    // auction batch size (offer denom amount)
+    pub batch_size: Uint128,
+    // auction phase configuration
+    pub auction_phase_config: AuctionPhaseConfig,
+    // amount of tokens required to be posted as a slashable bond
+    // in order to participate in the auction
+    pub solver_bond: Coin,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    /// adds an order to the auction to be executed on the next round.
+    /// only callable by orbital-core which is responsible for escrowing orders.
+    AddOrder(UserIntent),
+    /// finalizes the current auction round and prepares for the next
+    Tick { mock_fill_status: bool },
+    /// pause the auction, stopping any new orders from being accepted
+    Pause {},
+    /// resume the auction, allowing new orders to be accepted
+    Resume {},
+
+    // bidder actions
+    /// post a bond to participate in the auction
+    PostBond {},
+
+    /// withdraw the posted bond
+    WithdrawBond {},
```

Comment:
Not that important: Should we have an increase/decrease bond? If we are happy with Withdrawing and posting again It's fine